### DISCUSSION TITLE
功能讨论

### DISCUSSION THREAD

**Discussion by cyrushu** [EasyTier/EasyTier#374] [Ideas]
hi，作者您好，我想问一下咱们是否计划支持dns重载。感觉目前easytier的想法特别好，目前我们办公室的是easytier的方案。
我理解如果给办公室同事分配办公权限，是否应该用wireguard配置+dns配置以实现内网的域名解析？另外easytier本身是否计划dns重载？
因为我们最近有同事离职，所以我发现easytier在权限上没法做到单节点权限控制，如果是wireguard配置方案的话应该不会遇到这样的问题。

**Comment by cyrushu** [comment]
或者用预设密钥的方法实现权限控制的操作？

**Comment by KKRainbow** [comment]
关于权限控制，我的想法是这样的。
1. 支持以集群形式提供公共服务器功能，公共服务器本身形成一个虚拟网络，这个网络内部会配置一些子网代理。
2. 公共服务器可以新增、移除 RSA 密钥，其他节点可以使用公钥连入网络，相同公钥的节点组成一个虚拟网，并允许访问公共服务器集群代理的子网。
3. 密钥从集群撤销，就等于移除一个账号了

现在已经实现了公共服务器集群的能力（master 分支），后面一些功能属于企业级特性了，能做到，但是得看需求是否广泛

**Reply by cyrushu** [reply]
GoodPoint，有了集群性质的公共服务能力就可以给企业用户提供相对较完整的解决方案了，您可以从商业角度考虑一下。
目前国产的组网方案多多少少会遇到各种问题，配置不好就会强制走了公网，没有尝试直连之类的。你们做商业服务在带宽上的竞争力还是比较强的。
不过网络工具的项目还挺tricky的，希望easytier越做越好。

**Reply by KKRainbow** [reply]
我这个方案里，公共服务器是你们自建的。然后你们可以在这个集群里增删密钥来控制员工的权限。

员工可以把自己的虚拟网通过密钥接到你们的自建公共服务器，来访问你们公共服务器代理的子网

**Reply by KKRainbow** [reply]
至于商业角度，可以后面再考虑，目前还是把这个当成兴趣来做的。当然后续如果有商业合作我也欢迎

**Reply by cyrushu** [reply]
理解~我们需求太小了，目前自己做就能满足需求的。
不过如果建了商业集群的话，这个弹性伸缩的能力倒是挺有意思的，比较像vpn中的bt/pt下载。不过就会比较像国产的tailscale这样的产品定位了
